### DISCUSSION TITLE
Client state storage and management

### DISCUSSION THREAD

**Discussion by PasqualePuzio** [reflex-dev/reflex#1971] [General]
According to the docs

> Each user who opens your app has a unique ID and their own copy of the state. This means that each user can interact with the app and modify the state independently of other users.
All user state is stored on the server, and all event handlers are executed on the server. Reflex uses websockets to send events to the server, and to send state updates back to the client.

We were wondering: how is the state stored on the server? Is the state of a client removed at some point? If so, when (e.g. as soon as the session is closed)?

**Comment by masenf** [comment]
By default, the state is stored in a plain `dict`, keyed on "client_token", which is a UUID generated per session (browser tab). There is no expiration in this mode, which can lead to memory leaks and should really only be used during development.

When `REDIS_URL` is provided in the environment or in the `rxconfig.py`, then the state will be stored in redis with an expiration of 1 hour.

This can be configured, but isn't currently plumbed through into the configuration file. If you need to adjust the redis expiration, you can use the following in your main module after defining `rx.App`

```python
...
app = rx.App()
if isinstance(app.state_manager, rx.state.StateManagerRedis):
    app.state_manager.token_expiration = 7200
```
